src/libraries/System.Net.Sockets/tests/FunctionalTests/Connect.cs

```diff
@@ -204,4 +204,121 @@ public sealed class ConnectEap : Connect<SocketHelperEap>
+                cts.Cancel();
+
+                await Assert.ThrowsAnyAsync<OperationCanceledException>(async () => await client.ConnectAsync(new IPAddress[] { IPAddress.Parse("1.2.3.4"), IPAddress.Parse("1.2.3.5") }, 1, cts.Token));
+            }
+        }
+
+        [Fact]
+        public async Task ConnectHostNameAndPort_Precanceled_Throws()
+        {
+            using (var client = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            {
+                var cts = new CancellationTokenSource();
+                cts.Cancel();
+
+                await Assert.ThrowsAnyAsync<OperationCanceledException>(async () => await client.ConnectAsync("1.2.3.4", 1, cts.Token));
+            }
+        }
+
+        [Fact]
+        public async Task ConnectEndPoint_CancelDuringConnect_Throws()
+        {
+            EndPoint ep = new IPEndPoint(IPAddress.Parse("1.2.3.4"), 1);
+
+            using (var client = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            {
+                var cts = new CancellationTokenSource();
+
+                ValueTask t = client.ConnectAsync(ep, cts.Token);
+
+                cts.Cancel();
+
+                await Assert.ThrowsAnyAsync<OperationCanceledException>(async () => await t);
+            }
+        }
+
+        [Fact]
+        public async Task ConnectAddressAndPort_CancelDuringConnect_Throws()
+        {
+            using (var client = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            {
+                var cts = new CancellationTokenSource();
+
+                ValueTask t = client.ConnectAsync(IPAddress.Parse("1.2.3.4"), 1, cts.Token);
+
+                cts.Cancel();
+
+                await Assert.ThrowsAnyAsync<OperationCanceledException>(async () => await t);
+            }
+        }
+
+        [Fact]
+        public async Task ConnectMultiAddressAndPort_CancelDuringConnect_Throws()
+        {
+            using (var client = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            {
+                var cts = new CancellationTokenSource();
+
+                ValueTask t = client.ConnectAsync(new IPAddress[] { IPAddress.Parse("1.2.3.4"), IPAddress.Parse("1.2.3.5") }, 1, cts.Token);
+
+                cts.Cancel();
+
+                await Assert.ThrowsAnyAsync<OperationCanceledException>(async () => await t);
+            }
+        }
+
+        [Fact]
+        public async Task ConnectHostNameAndPort_CancelDuringConnect_Throws()
+        {
+            using (var client = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            {
+                var cts = new CancellationTokenSource();
+
+                ValueTask t = client.ConnectAsync("1.2.3.4", 1, cts.Token);
+
+                cts.Cancel();
+
+                await Assert.ThrowsAnyAsync<OperationCanceledException>(async () => await t);
```

Comment:
I doubt this will work on platforms other than Windows. See https://github.com/dotnet/runtime/pull/40506#discussion_r469968605

Reply:
Yep, will fix